designs/2022-forbid-disable/README.md

```diff
@@ -0,0 +1,125 @@
+- Repo: eslint/eslint
+- Start Date: 2022-02-28
+- RFC PR:
+- Authors: Brian Bartels
+
+# ESLintRC Option To Forbid Disable For Specific Rules
+
+## Summary
+
+Allow users to forbid users from disabling certain rules. This is similar to `noInlineConfig` where the primary difference is that `noInlineConfig` is set for all rules.
+
+## Motivation
+
+When working with complex lint configs. Some rules are more important than other. For this reason it can be inconsequential to disable certain rules. But incredibly bad to disable others.
+
+EG:
+```
+//eslint-disable-next-line @government/required-SECURITY-rule
+```
+Is a much worse rule to disable than
+```
+//eslint-disable-next-line noExtraSemi
+```
+
+`noInlineConfig` works well for some. But in reality, many users disable rules at least somewhat frequently for valid reasons. And in large repos, there are going to be exceptions that need inline configuration. 
+
+## Detailed Design
+
+In `eslintrc` I will add another option for rules. The new state can be
+```
+"warn"
+"error"
+"forbid"
+```
+where forbid is always an error and cannot be overridden by a directive
+
+EG `// eslint-disable semi` will have ~~semi~~ as an error, as well as the violation also having an error.
+
+Both errors will have a severity of 2. See [LintMessage#severity](https://eslint.org/docs/developer-guide/nodejs-api#-lintmessage-type)
+
+I will add a new array to the linter called `forbiddenRules` that will be processed on initialization of the linter.js. This will happen in the Constructor. Then in `getDirectiveComments` I will reference the rule being disabled in the directive. If that rule matches a rule in `forbiddenRules` then I will not accept that directive. And I will declare an error on that string. Note that this same logic applies to configuration comments as well.
```

Comment:
I think it would be better to encapsulate this in FlatConfigArray somehow. It already normalizes string severity levels to numbers so Linter doesn’t need to do that.

Reply:
I'm not sure how this will affect the design. The schema change will be related to the flatConfigArray. See if my updates made sense